src/wix/WixToolset.Core.Burn/Bundles/BurnWriter.cs

```diff
@@ -127,6 +130,11 @@ public bool AppendContainer(Stream containerStream, long containerSize, BurnComm
             UInt32 burnSectionCount = 0;
             UInt32 burnSectionOffsetSize = 0;
 
+            if (containerSize == 0)
```

Comment:
Investigate all of these checks for 0 byte containers. It seems like they were only necessary to catch when there was no attached container, but it should have checked the number of containers for that.